frame/support/procedural/src/partial_eq_no_bound.rs

```diff
@@ -0,0 +1,124 @@
+// This file is part of Substrate.
+
+// Copyright (C) 2020 Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/// Derive PartialEq but do not bound any generic.
+pub fn derive_partial_eq_no_bound(input: proc_macro::TokenStream) -> proc_macro::TokenStream {
+	use syn::spanned::Spanned;
+
+	let input: syn::DeriveInput = match syn::parse(input) {
+		Ok(input) => input,
+		Err(e) => return e.to_compile_error().into(),
+	};
+
+	let name = &input.ident;
+	let (impl_generics, ty_generics, where_clause) = input.generics.split_for_impl();
+
+	let impl_ = match input.data {
+		syn::Data::Struct(struct_) => match struct_.fields {
+			syn::Fields::Named(named) => {
+				let fields = named.named.iter()
+					.map(|i| i.ident.as_ref().expect("named fields have ident"))
+					.map(|i| quote::quote_spanned!(i.span() => self.#i == other.#i ));
+
+				quote::quote!( true #( && #fields )* )
+			},
+			syn::Fields::Unnamed(unnamed) => {
+				let fields = unnamed.unnamed.iter().enumerate()
+					.map(|(i, _)| syn::Index::from(i))
+					.map(|i| quote::quote_spanned!(i.span() => self.#i == other.#i ));
+
+				quote::quote!( true #( && #fields )* )
+			},
+			syn::Fields::Unit => {
+				quote::quote!( true )
+			}
+		},
+		syn::Data::Enum(enum_) => {
+			let variants = enum_.variants.iter()
+				.map(|variant| {
+					let ident = &variant.ident;
+					match &variant.fields {
+						syn::Fields::Named(named) => {
+							let names = named.named.iter()
+								.map(|i| i.ident.as_ref().expect("named fields have ident"));
+							let names_bis = names.clone()
```

Comment:
What does `bis` stands for?

Reply:
it is latin https://en.wikipedia.org/wiki/BIS
I guess we use it more in French than English, I changed it :-)